packages/web/src/services/artifacts.ts

```diff
@@ -2,15 +2,47 @@ import {getApiEndpoint} from '@services/apiEndpoint';
 
 import {getRtkBaseUrl, getRtkIdToken} from '@utils/rtk';
 
-const downloadFile = async (blob: Blob, fileName: string): Promise<void> => {
-  const url = URL.createObjectURL(blob);
-  const a = document.createElement('a');
-  a.href = url;
-  a.download = fileName;
-  document.body.appendChild(a);
-  a.click();
-  document.body.removeChild(a);
-  URL.revokeObjectURL(url);
+export const fetchContentFromSignedURL = async (signedURL: string): Promise<Blob> => {
+  const response = await fetch(signedURL);
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch content from signed URL: ${response.status} ${response.statusText}`);
+  }
+
+  return response.blob();
+};
+
+const getResponseFileUrlAndDownloadContent = async (response: Response, fileName: string) => {
+  if (!response.status || response.status >= 300) {
+    throw new Error('Something went wrong');
+  }
+
+  if (response.headers.get('content-type') === 'application/json') {
+    const data = await response.json();
+    const fileUrl = data.data?.url || data.url;
+    await downloadFile(fileUrl, fileName);
+    return;
+  }
+
+  const fileUrl = window.URL.createObjectURL(await response.blob());
+  await downloadFile(fileUrl, fileName);
+  URL.revokeObjectURL(fileUrl);
+};
+
+const downloadFile = async (signedURL: string, fileName: string): Promise<void> => {
+  try {
+    const content = await fetchContentFromSignedURL(signedURL);
```

Comment:
It would be much better to avoid the second fetch, as the User won't see the download progress bar otherwise, but it's minor at the moment - we will have soon a new BE endpoint, so we will be able to simplify that logic anyway - we won't need to check the JSON file for signed URL.